torchdynamo/eval_frame.py

```diff
@@ -51,6 +51,19 @@ def nothing():
 compile_lock = threading.Lock()
 
 
+def innermost_fn(fn):
+    """
+    In case of nesting of _TorchDynamoContext calls, find the innermost
+    function. TorchDynamo caches on fn.__code__ object, so its necessary to find
```

Comment:
@jansel, does this imply that if a user generates a closure, all separate instances of the closure will share the same cache? That seems suboptimal... the problem is especially prominent here, where `innermost_fn` is seemingly necessary but IMO it shouldn't be...

Reply:
Yes, multiple closures with the same `__code__` have the same cache.

Reply:
@jansel Ping if you missed this message.

Reply:
Perhaps we should emit a warning/error if you call `torchdynamo.optimize()` with two different compilers without calling `torchdynamo.reset()`.